third_party/move/move-compiler-v2/src/pipeline/explicit_drop.rs

```diff
@@ -30,16 +34,34 @@ impl FunctionTargetProcessor for ExplicitDrop {
         let mut transformer = ExplicitDropTransformer::new(target);
         transformer.transform();
         data.code = transformer.transformed;
-        data.annotations.remove::<LiveVarAnnotation>();
-        data.annotations.remove::<LifetimeAnnotation>();
-        data
+        self.recompute_invalidated_analyses(targets, fun_env, data, scc_opt)
     }
 
     fn name(&self) -> String {
         "ExplicitDrop".to_owned()
     }
 }
 
+impl ExplicitDrop {
+    /// Recomputes livevar analyais and lifetime analysis
+    fn recompute_invalidated_analyses(
+        &self,
+        targets: &mut FunctionTargetsHolder,
+        fun_env: &FunctionEnv,
+        mut data: FunctionData,
+        scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        // don't run the processor directly to avoid doing copy transformation again
+        let livevar_analysis_processor = LiveVarAnalysisProcessor();
+        let target = FunctionTarget::new(fun_env, &data);
+        let offset_to_live_refs =
+            LiveVarAnnotation::new(livevar_analysis_processor.analyze(&target));
+        data.annotations.set(offset_to_live_refs, true);
+        let reference_safety_processor = ReferenceSafetyProcessor {};
```

Comment:
Reference safety definitely not needed after this processor, livevar may. In any case, as discussed, perhaps make this dependent from an experiment flag, so we have the code ready for this.

Reply:
done